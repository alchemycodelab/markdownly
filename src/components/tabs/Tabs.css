.Tabs {
  display: flex;

  li {
    padding: 0 5px;
    margin: 5px;
    list-style: none;
    border: 1px solid slategray;
  }

}