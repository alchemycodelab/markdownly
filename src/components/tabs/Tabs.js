import React from 'react';
import PropTypes from 'prop-types';
import Tab from './Tab';
import styles from './Tabs.css';

const Tabs = ({ tabs, changeTabs, deleteTab }) => {
  const tabElements = tabs.map((tab, i) => {
    return (
      <li key={i}>
        <Tab title={tab} changeTabs={changeTabs} deleteTab={deleteTab} />
      </li>
    );
  });
  return (
    <ul className={styles.Tabs}>
      {tabElements}
    </ul>
  );
};

Tabs.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string),
  changeTabs: PropTypes.func,
  deleteTab: PropTypes.func
};

export default Tabs;
