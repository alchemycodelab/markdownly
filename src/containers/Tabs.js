import React from 'react';
import TabDeck from '../components/tab/TabDeck';
import { useSelector, useDispatch } from 'react-redux';
import { getSearchTerm, getMatchTitlesByTitle } from '../selectors/documentSelectors';
import { addTab, deleteTab, addSearch, getMatchTitle } from '../actions/documentActions';
import Filter from '../components/filter/Filter';

const Tabs = () => {
  const titles = useSelector(getMatchTitlesByTitle);
  const searchTerm = useSelector(getSearchTerm);
  const dispatch = useDispatch();
  const handleClick = tabNumber => dispatch(addTab(tabNumber));
  const handleDelete = title => dispatch(deleteTab(title));
  const handleSubmit = searchTerm => dispatch(getMatchTitle(searchTerm));
  const handleChange = ({ target }) => dispatch(addSearch(target.value));

  return (
    <>
      <TabDeck handleDelete={handleDelete} titles={titles} />
      <button onClick={() => handleClick(titles.length)}>ADD</button>
      <Filter handleSubmit={handleSubmit} handleChange={handleChange} searchTerm={searchTerm} />
    </>
  );
};

export default Tabs;


