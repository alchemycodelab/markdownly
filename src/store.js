import { createStore, compose } from 'redux';
import reducer from './reducers';
import { saveState } from './localStorage';


const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducer,
  composeEnhancers()
);

store.subscribe(() => {
  const state = store.getState();
  saveState({ state: state });
});

export default store;
