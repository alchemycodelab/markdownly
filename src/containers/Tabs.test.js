import React from 'react';
import { shallow } from 'enzyme';
import Tabs from './Tabs';
import store from '../store';

describe('Tabs component', () => {
  it('renders Tabs', () => {
    const wrapper = shallow(<Tabs store={store}/>);
    expect(wrapper).toMatchSnapshot();
  });
}); 
