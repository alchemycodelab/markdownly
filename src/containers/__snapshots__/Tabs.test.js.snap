// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders Tabs 1`] = `
<ContextProvider
  value={null}
>
  <Tabs
    handleClick={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    titles={
      Array [
        "default",
      ]
    }
  />
</ContextProvider>
`;
